frontend/compute-system/src/components/AppStore/components/AppCard.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button, Grid, Paper, Typography } from "@material-ui/core";
+import { PanoramaOutlined } from "@material-ui/icons";
+import * as React from "react";
+import appCardStyles from "./appCardStyles";
+
+interface AppCardProps {
+  title: string;
+  updatedDate: string;
+  description: string;
+  timesUsed: number;
+  rate: number;
+}
+
+const AppCard = (props: AppCardProps) => {
+  const classes = appCardStyles();
+
+  return (
+    <Paper className={classes.paper}>
+      <Grid container alignItems="center" wrap="nowrap">
+        <Grid item>
+          <PanoramaOutlined className={classes.icon} />
+        </Grid>
+        <Grid
+          item
+          container
+          direction="column"
+          wrap="nowrap"
+          className={classes.textBox}
+        >
+          <Grid item>
+            <Typography variant="h5">{props.title}</Typography>
+          </Grid>
+          <Grid item className={classes.updatedText}>
+            <Typography variant="body1">{`Updated on ${props.updatedDate}`}</Typography>
+          </Grid>
+          <Grid item>
+            <Typography variant="caption">{props.description}</Typography>
+          </Grid>
+        </Grid>
+        <Grid item container direction="column" alignItems="center">
+          <Grid item container wrap="nowrap" className={classes.usedRateBox}>
+            <Grid item container direction="column" alignItems="center">
+              <Grid item>
+                <Typography variant="subtitle2">{props.timesUsed}</Typography>
+              </Grid>
+              <Grid item>
+                <Typography variant="caption">Used</Typography>
+              </Grid>
+            </Grid>
+            <Grid item container direction="column" alignItems="center">
+              <Grid item>
+                <Typography variant="subtitle2">{props.rate}</Typography>
+              </Grid>
+              <Grid item>
+                <Typography variant="caption">Rate</Typography>
+              </Grid>
+            </Grid>
+          </Grid>
+          <Grid item>
+            <Button className={classes.addButton}>Add to cockpit</Button>
+          </Grid>
+        </Grid>
+      </Grid>
+    </Paper>
```

Comment:
a little messy... could think of breaking this into multiple small components

Reply:
true, but want to xD